pac4j-core/src/main/java/org/pac4j/core/client/AnonymousClient.java

```diff
@@ -0,0 +1,68 @@
+/*
+  Copyright 2012 - 2015 pac4j organization
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+ */
+package org.pac4j.core.client;
+
+import org.pac4j.core.context.WebContext;
+import org.pac4j.core.credentials.AnonymousCredentials;
+import org.pac4j.core.exception.RequiresHttpAction;
+import org.pac4j.core.exception.TechnicalException;
+import org.pac4j.core.profile.AnonymousProfile;
+
+/**
+ * Anonymous client. Not to be used except for advanced use cases.
+ *
+ * @author Jerome Leleu
+ * @since 1.8.1
+ */
+public class AnonymousClient extends BaseClient<AnonymousCredentials, AnonymousProfile> {
```

Comment:
To secure the /status and /statistics endpoints of the CAS server, I use the `IpClient` with an `IpRegexpAuthenticator`: from a pac4j point of view, the user is not authenticated if he has not the right IP. Thus, a 401 HTTP error. Spring Security returns a 403 HTTP error.
To fully mimic the Spring Security behaviour, the solution at the pac4j level is to use an `AnonymousClient` (the user is authenticated even if anonymous) and an `IpRegexpAuthorizer` to generate a 403 HTTP error in case of improper IP.